tools/devops/automation/scripts/GitHub.psm1

```diff
@@ -1,3 +1,41 @@
+<#
+    .SYNOPSIS
+        Simple retyr block to workaorund certain issues with the webservices that cannot handle the load.
+
+    .PARAMETER Request
+        The request to be performed and retired if failed.
+
+    .PARAMETER Retries
+        The number of times the we will retry to perform the request.
+#>
+function Invoke-Request {
+    param (
+        [scriptblock]
+        $Request,
+
+        [int]
+        $Retries=5
+    )
+    $count = 0
+    do {
+        try {
+            # that & is important, tells pwsh to execute the script block, else you simple returns the block itself
+            return & $Request
+        } catch {
+            if ($count -gt $Retries) {
+                # notify and throw
+                Write-Host "Could not perform request after $Retries attempts."
+                throw $_.Exception
+            } else {
+                Write-Host "Error performing request trying in 30s"
+                Start-Sleep -Seconds 30
+                $count = $count + 1
+            }
+        }
+
+    } while ($true)
```

Comment:
No need to set any var, if we are bigger than $Retries we are throwing, if we are smaller we return. Thanks @rolfbjarne for the comment.